app/services/trend_importer.rb

```diff
@@ -32,17 +32,34 @@ def update_trend_counters
       end
     end
     Trend.has_value.each do |trend|
+      new_count = if trend.is_a? FactTrend
+                    counter_hash[trend.trendable_id][trend.fact_value]
+                  else
+                    counter_hash[trend.trendable_type][trend.trendable_id]
+                  end || 0
+
+      latest_counter = trend.trend_counters.order(:created_at).last
+      if latest_counter
+        latest_counter.interval_end = timestamp
```

Comment:
if the latest counter has changed, should not we keep the value of the last time the count was measured ? changing it to the current timestamp is odd if the value already changed?